src/test/java/de/tum/in/www1/artemis/util/GitUtilService.java

```diff
@@ -99,26 +101,40 @@ private Repository initializeRepo(Path path) throws IOException {
     }
 
     public void deleteRepos() {
-        try {
-            if (remoteRepo != null) {
-                remoteRepo.close();
+        if (remoteRepo != null) {
+            remoteRepo.close();
+        }
+        if (localRepo != null) {
+            localRepo.close();
+        }
+        if (localGit != null) {
+            localGit.close();
+        }
+        if (remoteGit != null) {
+            remoteGit.close();
+        }
+        while (FileUtils.isDirectory(localPath.toFile())) {
```

Comment:
this could end up in an endless loop.
I suggest that you wait a short time in case it fails and try it at most 10 times.